src/ubkg_api/app.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+from pathlib import Path
+
+from flask import Flask, jsonify
+
+from ubkg_api.neo4j_manager import Neo4jManager
+from ubkg_api.routes.assaytype.assaytype_controller import assaytype_blueprint
+from ubkg_api.routes.codes.codes_controller import codes_blueprint
+from ubkg_api.routes.concepts.concepts_controller import concepts_blueprint
+from ubkg_api.routes.datasets.datasets_controller import datasets_blueprint
+from ubkg_api.routes.organs.organs_controller import organs_blueprint
+from ubkg_api.routes.semantics.semantics_controller import semantics_blueprint
+from ubkg_api.routes.terms.terms_controller import terms_blueprint
+from ubkg_api.routes.tui.tui_controller import tui_blueprint
+from ubkg_api.routes.valueset.valueset_controller import valueset_blueprint
+
+logging.basicConfig(format='[%(asctime)s] %(levelname)s in %(module)s: %(message)s', level=logging.DEBUG,
+                    datefmt='%Y-%m-%d %H:%M:%S')
+logger = logging.getLogger(__name__)
+
+
+class UbkgAPI:
+    def __init__(self, config):
+
+        self.app = Flask(__name__, instance_path=os.path.join(os.path.abspath(os.path.dirname(__file__)), '../instance'), instance_relative_config=True)
+
+        self.app.register_blueprint(assaytype_blueprint)
+        self.app.register_blueprint(codes_blueprint)
+        self.app.register_blueprint(concepts_blueprint)
+        self.app.register_blueprint(datasets_blueprint)
+        self.app.register_blueprint(semantics_blueprint)
+        self.app.register_blueprint(tui_blueprint)
+        self.app.register_blueprint(valueset_blueprint)
+        self.app.register_blueprint(terms_blueprint)
+        self.app.register_blueprint(organs_blueprint)
+
+        neo4j = None
+
+        try:
+            if Neo4jManager.is_initialized():
+                neo4j = Neo4jManager.instance()
+                logger.info("Neo4jManager has already been initialized")
+            else:
+                if not config:
+                    logger.info('Config not provided. Looking for app.cfg ...')
+                    self.app.config.from_pyfile('app.cfg')
+                    neo4j = Neo4jManager.create(self.app.config['SERVER'], self.app.config['USERNAME'], self.app.config['PASSWORD'])
```

Comment:
@yuanzhou I worked with @SamSedivy to rework this PR to model the Search API/ Search Adapter with main.py being the primary entry point into the ubkg_api PyPI package. Would like your opinion on handling configuration here though

Reply:
@maxsibilla @SamSedivy currently the UBKG-API serves as a query interface to return the ontology data from Neo4j, and it doesn't handle additional business logic. So this configuration is fine. If later we'll want the UBKG-API to do more things (Globus interactions or configurable AWS resources and some HuBMAP/SenNet application-specific settings), the search-api/search-adaptor configuration pattern will be a better design to separate the configuration concerns.